conan/internal/api/new/autoools_exe.py

```diff
@@ -43,6 +43,13 @@ def build(self):
     def package(self):
         autotools = Autotools(self)
         autotools.install()
+
+    {% if requires is defined -%}
```

Comment:
I'll try to go beyond this:
- Add ``find_package()`` or equivalent to build scripts
- Add a ``#include "<name>.h"`` to code
- Add a call ``<name>();`` to code

That is, make it a fully functional compiling and linking requires (that cannot be made automatic for libs like zlib, but it is a template, you should replace the call with ``zlib()`` to something that makes sense. 

Maybe not for all build systems, but maybe for CMake?

Reply:
Nice idea

Reply:
After looking into this just now, it's a bit less straightforward than I originally thought, let's do this in a different PR :)